R/conditions.R

```diff
@@ -75,6 +75,14 @@ getCallNames <- function(calls) {
   })
 }
 
+# Faster version of getCallNames, whilst maintaining same structure
+getCallNamesForHash <- function(calls) {
+  lapply(calls, function(call) {
+    name <- call[[1L]]
+    is.function(name) || typeof(name) == "promise" || return(name)
```

Comment:
```suggestion
    if (is.function(name)) return("<Anonymous>")
    if (typeof(name) == "promise")) return("<Promise>")
    name
```

Reply:
Yes, I'm fine with these, thanks! Seems safer - I was probably being too aggressive with the optimization, and practically it doesn't really make a difference.